qa/rgw/store/sfs/tests/sfs-smoke-test.sh

```diff
@@ -44,101 +41,158 @@ signature_v2 = True
 signurl_use_https = False
 EOF
 
-pushd ${testpath}
+  pushd "${testpath}" > /dev/null || exit 1
+}
+
+cleanup() {
+  popd > /dev/null || exit 1
+
+  rm -rf "${testpath}"
+}
+
+success() {
+  [[ "$(caller 1 | cut -d' ' -f2)" == "main" ]] && \
+    echo -e "$(caller 1) \e[0;32mSUCCESS\e[0m"
+}
+
+failure() {
+  [[ "$(caller 1 | cut -d' ' -f2)" == "main" ]] && \
+    echo -e "$(caller 1) \e[1;31mFAILURE\e[0m"
+  retcode="1"
+}
+
+# this function is meant to be called indirectly from expect_success
+# it will ignore a failure, but not fail on success, unlike expect_failure,
+# which will fail on success
+# shellcheck disable=SC2317
+ignore_failure() {
+  "$@" || true
+}
+
+expect_failure() {
+  "$@" > /dev/null 2>&1 && failure
+  success
```

Comment:
aren't we always outputting `success` here anyways, even if `failure` is hit?

Reply:
Yes. Good catch, this logic is fubar.